torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1128,19 +1128,42 @@ def safe_expand(r):
     else:
         return r
 
+cast_bool_to_int_magic_methods = {
+    'add',
+    'sub',
+    'mul',
+    'pow',
+    'mod',
+    'truediv',
+    'floordiv',
+    'rshift',
+    'lshift',
+    'abs',
+    'ceil',
+    'floor',
+    'neg',
+    'sym_sqrt',
+}
+
+def b2i(sympy_val):
+    if isinstance(sympy_val, sympy.logic.boolalg.Boolean) and not isinstance(sympy_val, sympy.Symbol):
```

Comment:
Another hacky bit which I'm unsure will hold in all cases.

Reply:
can a Boolean be a symbol at the same time??

Reply:
Also, I think that the right way of doing this goes through straight casting it to `sympy.Integer`.
cc @asmeurer

Reply:
`sympy.Integer` would only work for constant values. Also AFAICT there is no symbolic equivalent which is annoying.

Reply:
> can a Boolean be a symbol at the same time??

Somehow, as classes, a boolean is a symbol but a symbol is not a boolean.

Reply:
Yes unfortunately SymPy's Symbol object currently represents both boolean and numerical expressions. There's an open issue about this in SymPy, but it's not straightforward to fix because of backwards compatibility https://github.com/sympy/sympy/issues/8192. Unless PyTorch is explicitly using Symbol as a boolean (e.g., to construct some boolean predicates), it shouldn't ever be one in PyTorch contexts.